### ISSUE TITLE
Test runs with a single example fail

### ISSUE THREAD

**Issue by RAMitchell** [rapidsai/legate-boost#41]
When running

`legate --module pytest legateboost/test/test_with_hypothesis.py -svk 'regressor'`

we see some hypothesis test failures such as 

```
Falsifying example: test_regressor(
    model_params={'n_estimators': 1,
     'max_depth': 1,
     'init': None,
     'random_state': 0},
    regression_params={'objective': 'squared_error', 'learning_rate': 1.0},
    regression_dataset=(array([[0.436]], dtype=float16),
     array([[0.02592]], dtype=float16),
     None),
)
```
where the dataset contains a single element. The store is backed by a future. I printed out the values of gradients (in this case a Store of size 1 backed by a future) inside the cpu tree building task and saw that they would randomly change between 0 and the correct values on different runs. 

This can be worked around for now by not generating datasets with a single row https://github.com/rapidsai/legateboost/blob/86bbb028442fe2c0dd96376de7b711ae60ea9446/legateboost/test/test_with_hypothesis.py#L59
